cmd/sos_sync.go

```diff
@@ -0,0 +1,518 @@
+					} else {
+						bufferSize = task.Size
+					}
+
+					file, err := os.Open(localPath)
+					if err != nil {
+						errorChannel <- err
+					} else {
+						buffer := make([]byte, bufferSize)
+						_, err = file.Read(buffer)
+						if err != nil {
+							taskUi.error()
+							errorChannel <- err
+						} else {
+							contentType = http.DetectContentType(buffer)
+							//endregion
+
+							//region Upload
+							if !sosSyncConfiguration.DryRun {
+								f, err := os.Open(localPath)
+								if err != nil {
+									taskUi.error()
+									errorChannel <- err
+								} else {
+									//noinspection GoUnhandledErrorResult
+									defer f.Close()
+
+									_, upErr := sosClient.PutObjectWithContext(
+										gContext,
+										sosSyncConfiguration.TargetBucket,
+										remotePath,
+										taskUi.getReader(f),
+										task.Size,
+										minio.PutObjectOptions{
+											ContentType: contentType,
+										},
+									)
+									if upErr != nil {
+										taskUi.error()
+										errorChannel <- upErr
+									} else {
+										taskUi.complete()
+									}
+								}
+							} else {
+								taskUi.complete()
+							}
+							//endregion
+						}
+					}
+				}
+				<-workerSem
+			}()
+		}
+
+		//Wait for complete input before waiting for the task WG
+		<-inputDone
+		close(inputDone)
+		taskWG.Wait()
+	}
+}
+
+//Takes a preconfigured object storage client for the target bucket, and a validated configuration
+//and synchronizes the local folder with the remote.
+func sosSyncProcess(
+	sosSyncConfiguration sosSyncConfiguration,
+	sosSyncDiff sosSyncDiff,
+	sosSyncProcessTaskList sosSyncProcessTaskList,
+) error {
+	done := make(chan bool, 1)
+	errorChannel := make(chan error)
+	taskList := sosSyncDiff(sosSyncConfiguration, done, errorChannel)
+
+	sosSyncProcessTaskList(sosSyncConfiguration, taskList, done, errorChannel)
+	var err error = nil
+	for {
+		select {
+		case errChannelResult := <-errorChannel:
+			//noinspection GoUnhandledErrorResult
+			fmt.Fprintf(os.Stderr, "Error while processing: %s", errChannelResult)
+			err = errChannelResult
+		default:
+			err = nil
+		}
+		if err == nil {
+			break
+		}
+	}
+	if //noinspection GoNilness
+	err != nil {
+		return errors.New("one or more errors happened while processing your sync")
+	} else {
+		return nil
+	}
+}
+
+//endregion
+
+//region Wiring
+
+//Creates the live SOS client
+func sosSyncLiveClientFactory(certsFile string) (*sosClient, error) {
+	client, err := newSOSClient(certsFile)
+	if err != nil {
+		return nil, err
+	}
+	return client, err
+}
+
+//Initializes the sync command
+func init() {
+	runE := newSosSyncRunE(sosSyncLiveClientFactory)
+	cmd := &cobra.Command{
+		Use:     "sync <bucket name> <local path> <remote-path>",
+		Short:   "Sync a local folder with the object storage",
+		Aliases: gSyncAlias,
+		RunE:    runE,
+	}
+	cmd.Flags().BoolP("remove-deleted", "r", false, "Remove remote files not present locally")
```

Comment:
I suggest using the Rsync terminology, as many people are familiar with it:

```suggestion
	cmd.Flags().BoolP("remove-deleted", "d", false, "Delete remote files not present locally")
```